src/containers/CreateNote/CreateNote.js

```diff
@@ -14,90 +16,90 @@ export class CreateNote extends React.Component {
       noteItems: this.props.noteItems || [],
       currentFocus: null
     };
-  }
+  };
 
   handleChangeTitle = event => {
     this.setState({ title: event.target.value, currentFocus: null });
   };
 
-  makeCopy = element => JSON.parse(JSON.stringify(element))
-  
   handleChangeNoteItems = event => {
-    const noteItemsCopy = this.makeCopy(this.state.noteItems);
+    const noteItemsCopy = this.makeCopy();
+    const { id } = event.target.closest('label');
     const { value: newText } = event.target;
-    const { id } = event.target.closest('label')
-    const matchedNoteItem = noteItemsCopy.find(note => note.id === id);
-
+    const matchedNoteItem = this.findMatchingNoteItem(noteItemsCopy, id);
     if (matchedNoteItem) {
-      matchedNoteItem.text = newText;     
+      matchedNoteItem.text = newText;
     } else {
       const newListItem = { id, text: newText, isCompleted: false };
       noteItemsCopy.push(newListItem);
     }
+    this.updateNoteItems(noteItemsCopy, id);
+  };
 
-    this.setState({
-      noteItems: noteItemsCopy,
-      currentFocus: id
-    });
+  handleItemDelete = event => {
+    const noteItemsCopy = this.makeCopy();
+    const { id } = event.target.closest('label');
+    const noteItemIndex = noteItemsCopy.findIndex(note => note.id === id);
+    noteItemsCopy.splice(noteItemIndex, 1);
+    this.updateNoteItems(noteItemsCopy, id);
   };
 
   handleToggleIsComplete = event => {
-    const noteItemsCopy = this.makeCopy(this.state.noteItems);
+    const noteItemsCopy = this.makeCopy();
     const { id } = event.target.closest('label');
-    const matchedNoteItem = noteItemsCopy.find(note => note.id === id);
+    const matchedNoteItem = this.findMatchingNoteItem(noteItemsCopy, id);
+    matchedNoteItem.isCompleted = !matchedNoteItem.isCompleted;
+    this.updateNoteItems(noteItemsCopy, id);
+  };
 
-    if(matchedNoteItem) {
-      matchedNoteItem.isCompleted = !matchedNoteItem.isCompleted
-    } else {
-      const newListItem = {id, text: '', isCompleted: event.target.checked}
-      noteItemsCopy.push(newListItem);
-    }
+  makeCopy = () => JSON.parse(JSON.stringify(this.state.noteItems));
```

Comment:
Could we rename this to makeCopyNoteItems (or similar) since it's scope has been brought down to just making a copy of noteItems?

Reply:
Yeah:) sounds good to me. I'll change it to 'makeNoteItemsCopy' 👍